src/script/conversation/ConversationRepository.ts

```diff
@@ -2140,9 +2122,9 @@ export class ConversationRepository {
    * @param conversation Conversation to clear content from
    * @param timestamp Timestamp of the event
    */
-  public async clearConversation(conversation: Conversation, timestamp?: number) {
+  public async clearConversation(conversation: Conversation) {
     await this.messageRepository.updateClearedTimestamp(conversation);
-    return this.clearConversationContent(conversation, timestamp);
+    return this.clearConversationContent(conversation, new Date().getTime());
```

Comment:
actually, can you check something for me. 

Since here we send the `clearConversation` event through the wire to all the devices, does the current device also receives it? In this case we might not even need to call `this.clearConversationContent` at this point (since the injected event will trigger the right function calls). 

Am I right?

Reply:
Just double checked, no the self client (client that sends the message) does not receive the message. Self client is the one that sends the message in a self-conversation (proteus self or mls self or both), so it's not targeted as a recipient. We need to run the method manually for the self client.